src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadPropertiesStepExecution.java

```diff
@@ -82,6 +87,9 @@
             properties.load(sr);
         }
 
+        // Interpolated values in the properties
+        properties = interpolateProperties(properties);
```

Comment:
I do not recommend changing the behavior of the `readProperties` step. Rather introduce a new step for interpolation if you want one. Or just leave this to a global library with `@Grab` for people who need it.

Reply:
Or maybe adding a new parameter telling the step to change behavior?

Reply:
Sounds better with a parameter, the impact on the code is minimal.